lxqt-leave/leavedialog.cpp

```diff
@@ -51,6 +55,25 @@ LeaveDialog::LeaveDialog(QWidget* parent)
     setWindowFlags((Qt::CustomizeWindowHint | Qt::FramelessWindowHint |
                     Qt::WindowStaysOnTopHint | Qt::X11BypassWindowManagerHint));
 
+    // set the layer and centered position under Wayland
+    if (QGuiApplication::platformName() == QStringLiteral("wayland")) {
+        winId();
+        if(QWindow* win = windowHandle()) {
+            if(LayerShellQt::Window* layershell = LayerShellQt::Window::get(win)) {
+                layershell->setLayer(LayerShellQt::Window::Layer::LayerOverlay);
+                layershell->setKeyboardInteractivity(LayerShellQt::Window::KeyboardInteractivityExclusive);
+                LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorTop};
+                layershell->setAnchors(anchors);
+                QScreen *screen = win->screen();
+                if (screen == nullptr)
+                    screen = QGuiApplication::primaryScreen();
+                QRect desktop = screen->availableGeometry();
+                int topMargin = desktop.center().y() - sizeHint().height();
```

Comment:
As I'd suspected, there's a code problem here.

Logically, we should have used `sizeHint().height() / 2`, but what `sizeHint()` reports inside the c-tor isn't the real size of the dialog. _Only by chance, the reported `sizeHint().height()` is almost half the real `sizeHint().height()`,_ and that's why the dialog looks centered under Wayland.

To have a correct code, we can do this inside `showEvent()`. IMHO, it may be better to do it after 2.0.0.